Adds support for sending custom IMAP command

libec opened the pull request:
Also closes #1247

libec commented:
I'd say it fails because mailcore first tries to download prebuilt libetpan, right?

dinhviethoa commented:
It think that's the issue. I'll update it this week-end.
Meanwhile, could you update the CMake build file? It should be easy enough even if you don't know yet about CMake. It's mainly in src/cmake.

libec commented:
Sure, will do tomorrow. Have to go afk right now.

dinhviethoa commented on src/core/imap/MCIMAPSession.cpp:
```diff
@@ -996,6 +996,13 @@ static uint64_t get_mod_sequence_value(mailimap * session)
     return mod_sequence_value;
 }
 
+String * IMAPSession::sendCustomCommand(String * command)
```
s/sendCustomCommand/customCommand/
Also, you need to add an ErrorCode parameter.

libec commented:
I've added the `ErrorCode` as parameter to `customCommand` calls, please let me know if you think it's handled properly or needs some more work. I've also changed the name of the functions from `sendCustomCommand` to `customCommand`.

dinhviethoa commented on src/core/imap/MCIMAPSession.cpp:
```diff
@@ -996,6 +996,20 @@ static uint64_t get_mod_sequence_value(mailimap * session)
+    int r;
+    
+    r = mailimap_custom_command(mImap, MCUTF8(command));
+    if (r == MAILIMAP_ERROR_CUSTOM_COMMAND) {
```
Could you change the error management to the following?

```
    if (r == MAILIMAP_ERROR_STREAM) {
        mShouldDisconnect = true;
        * pError = ErrorConnection;
        return;
    }
    else if (r == MAILIMAP_ERROR_PARSE) {
        * pError = ErrorParse;
        return;
    }
    else if (hasError(r)) {
        * pError = ErrorDelete;
        return;
    }
```

dinhviethoa commented:
One comment then we're good to go.

dinhviethoa commented:
I'll fixed it after merge.

libec commented:
Thanks for your guidance :+1: Great work on libetpan and mailcore btw. !

dinhviethoa commented:
Thanks for your patch!
